zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,413 @@ where
+/// after `height`.
+async fn load_transactions_from_block_after(
+    height: block::Height,
+    network: Network,
+    state_path: &Path,
+) -> Result<Vec<Arc<Transaction>>> {
+    let (_read_write_state_service, mut state, latest_chain_tip, _chain_tip_change) =
+        start_state_service(network, state_path.join("state")).await?;
+
+    let tip_height = latest_chain_tip
+        .best_tip_height()
+        .ok_or_else(|| eyre!("State directory doesn't have a chain tip block"))?;
+
+    assert!(
+        tip_height > height,
+        "Chain not synchronized to a block after the specified height"
+    );
+
+    let mut target_height = height.0;
+    let mut transactions = Vec::new();
+
+    while transactions.is_empty() {
+        transactions =
+            load_transactions_from_block(block::Height(target_height), &mut state).await?;
+
+        transactions.retain(|transaction| !transaction.has_any_coinbase_inputs());
+
+        target_height += 1;
+    }
+
+    Ok(transactions)
+}
+
+/// Performs a request to the provided read-only `state` service to fetch all transactions from a
+/// block at the specified `height`.
+async fn load_transactions_from_block<ReadStateService>(
+    height: block::Height,
+    state: &mut ReadStateService,
+) -> Result<Vec<Arc<Transaction>>>
+where
+    ReadStateService: Service<
+        zebra_state::ReadRequest,
+        Response = zebra_state::ReadResponse,
+        Error = zebra_state::BoxError,
+    >,
+{
+    let request = zebra_state::ReadRequest::Block(HashOrHeight::Height(height));
+
+    let response = state
+        .ready()
+        .and_then(|ready_service| ready_service.call(request))
+        .map_err(|error| eyre!(error))
+        .await?;
+
+    let block = match response {
+        zebra_state::ReadResponse::Block(Some(block)) => block,
+        zebra_state::ReadResponse::Block(None) => {
+            panic!("Missing block at {height:?} from state")
+        }
+        _ => unreachable!("Incorrect response from state service: {response:?}"),
+    };
+
+    Ok(block.transactions.to_vec())
+}
+
+/// Starts a state service using the provided `cache_dir` as the directory with the chain state.
+async fn start_state_service(
+    network: Network,
+    cache_dir: impl Into<PathBuf>,
+) -> Result<(
+    BoxStateService,
+    impl Service<
+        zebra_state::ReadRequest,
+        Response = zebra_state::ReadResponse,
+        Error = zebra_state::BoxError,
+    >,
+    LatestChainTip,
+    ChainTipChange,
+)> {
+    let config = zebra_state::Config {
+        cache_dir: cache_dir.into(),
+        ..zebra_state::Config::default()
+    };
+
+    Ok(zebra_state::init(config, network))
+}
+
+/// Spawns a zebrad instance to interact with lightwalletd, but without an internet connection.
+///
+/// This prevents it from downloading blocks. Instead, the `zebra_directory` parameter allows
+/// providing an initial state to the zebrad instance.
+fn spawn_zebrad_for_rpc_without_initial_peers(
```

Comment:
Optional: We could move this utility function to one of the test sub-modules.

Reply:
I'm not sure where to move this to, and I'm not sure if this is too specific to this test :thinking: 

Should I:

1. Move it to `launch`?
2. Move it to `cached_state`?
3. Keep it here?

Reply:
Seems like it could be useful for a lot of tests?
Some tests we won't want Zebra updating the tip while we're testing.

Maybe `launch`?

Reply:
Done.